src/gpl/src/solver.h

```diff
@@ -0,0 +1,79 @@
+#ifndef __SOLVER__
+#define __SOLVER__
+
+#include <Eigen/IterativeLinearSolvers>
+#include <Eigen/SparseCore>
+#include <memory>
+
+#ifdef ENABLE_GPU
+#include "gpuSolver.h"
+#endif
+#include "graphics.h"
+#include "odb/db.h"
+#include "placerBase.h"
+#include "plot.h"
+#include "utl/Logger.h"
+
+namespace utl {
+class Logger;
+}
+
+namespace gpl {
+
+struct err{
+  float errorX; // The relative residual error for X
+  float errorY; // The relative residual error for Y
+};
+
+using Eigen::BiCGSTAB;
+using Eigen::IdentityPreconditioner;
+using utl::GPL;
+
+typedef Eigen::SparseMatrix<float, Eigen::RowMajor> SMatrix;
+
+#ifdef ENABLE_GPU
+ err cudaSparseSolve(int iter,
```

Comment:
This is too long to inline in a header.  Same below.

Reply:
Do you suggest that I have a solver.cpp correspondingly?

Reply:
Yes